### PR TITLE
enhance pipeline logic per build type

### PR DESCRIPTION
**Pull request by sumansreenivasan**
### Change description ###

enhance pipeline logic per build type + befta fw upgrade with added imports of definitions

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: Jenkinsfile_CNP [hunk 1]
```diff
@@ -48,24 +48,29 @@ def vaultOverrides = [
     'spreview': 'saat'
 ]
 
-// vars needed for functional tests
+// Env variables to switch pipeline logic and wiring per type of build
+env.DATA_STORE_DEVELOP_PR       = "pr-601" // This doesn't change frequently, but when it does, only change this value.
+env.DEFINITION_STORE_DEVELOP_PR = "pr-575" // This doesn't change frequently, but when it does, only change this value.
+env.ACTIVITY_API_DEVELOP_PR     = "PR-211" // This doesn't change frequently, but when it does, only change this value.
+env.PRs_TO_USE_AAT              = "PR-217" // Set this value to a PR number, or add it as a comma-separated value, if it's to follow CI/CD.
```
Comment (danlysiak) [inline Jenkinsfile_CNP:55]: Maybe worth adding to the comment that this one importantly needs to always be uppercase, to work with L#84?

Comment (MSancaktutar) [reply Jenkinsfile_CNP:55]: made case-insensitive.

File: Jenkinsfile_CNP [hunk 2]
```diff
@@ -48,24 +48,29 @@ def vaultOverrides = [
     'spreview': 'saat'
 ]
 
-// vars needed for functional tests
+// Env variables to switch pipeline logic and wiring per type of build
+env.DATA_STORE_DEVELOP_PR       = "PR-601" // This doesn't change frequently, but when it does, only change this value.
```
Comment (mario-paniccia) [inline Jenkinsfile_CNP:52]: I think you could use a normal variable here? i.e.
def DATA_STORE_DEVELOP_PR = "PR-601"
see
https://github.com/hmcts/ccd-data-store-api/blob/cbdbb95e62ff2fdf9acfa695fd4710ab51bfa81d/Jenkinsfile_CNP#L14

Comment (MSancaktutar) [reply Jenkinsfile_CNP]: Sure, good point.

**Pull request merged**